### PR TITLE
sql: add replication start time to `SHOW TENANT WITH REPLICATION STATUS`

### PR DESCRIPTION
**Pull request by adityamaru**
The start time is a useful piece of information to expose to the end user. It is the lower bound for the data we have replicated.

Informs: #93447

Release note: None

### CODE REVIEW

**Review by lidorcarmel — APPROVED** [review]
lgtm!

File: pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go [hunk 1]
```diff
@@ -1278,6 +1282,6 @@ func TestTenantStreamingShowTenant(t *testing.T) {
 	require.Less(t, maxReplTime, timeutil.Now())
 	require.Less(t, protectedTime, timeutil.Now())
 	require.GreaterOrEqual(t, maxReplTime, highWatermark.GoTime())
-	// TODO(lidor): replace this start time with the actual replication start time when we have it.
-	require.GreaterOrEqual(t, protectedTime, testStartTime)
+	require.GreaterOrEqual(t, protectedTime, replicationDetails.ReplicationStartTime.GoTime())
+	require.Equal(t, replicationStartTime.UnixNano(), replicationDetails.ReplicationStartTime.GoTime().UnixNano())
```
Comment (lidorcarmel) [inline pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go:1286]: can you drop the `UnixNano()`?

**Review by lidorcarmel — APPROVED** [review]

File: pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go (hunk 1 above)
Comment (adityamaru) [reply pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go]: The test flake actually revealed that the `replicationStartTime` is only accurate up to the microsecond after we start running into rounding mismatches. I had to change this to `UnixMicro` to pass under stress on a gceworker.

**Comment by adityamaru** [comment]
Tracking the flake here https://github.com/cockroachdb/cockroach/issues/93731

bors r=lidorcarmel

**Pull request merged**